datafusion/src/datasource/object_store/mod.rs

```diff
@@ -33,6 +33,12 @@ use local::LocalFileSystem;
 
 use crate::error::{DataFusionError, Result};
 
+/// Both Read and Seek
+pub trait ReadSeek: Read + Seek {}
+
+impl<R: Read + Seek> ReadSeek for std::io::BufReader<R> {}
+impl<R: AsRef<[u8]>> ReadSeek for std::io::Cursor<R> {}
```

Comment:
Not sure the trait is needed, but 

```suggestion
impl<T: Read + Seek> ReadSeek for T {}
```

should generalize for everything.

Reply:
Yes they are not needed, I originally had just Read + Seek in my branch because BufReader is Seek if R is Seek and Cursor is Seek on &[u8] in std. Should be removed.

Reply:
Nice tip on the generic approach. @Igosuki is right that this is not needed for BufRead and Cursor, It's only needed for `std::fs::File`, so I will keep the generic form.